src/partial-result-stream.ts

```diff
@@ -187,20 +196,57 @@ export class PartialResultStream extends Transform implements ResultEvents {
         .fields as google.spanner.v1.StructType.Field[];
     }
 
+    let res = true;
     if (!is.empty(chunk.values)) {
-      this._addChunk(chunk);
+      res = this._addChunk(chunk);
     }
 
-    next();
+    if (res || !this._requestStream) {
+      next();
+    } else {
+      // Wait a little before we push any more data into the pipeline as a
+      // component downstream has indicated that a break is needed. Pause the
+      // request stream to prevent it from filling up the buffer while we are
+      // waiting.
+      setTimeout(() => {
```

Comment:
`2` and `4` are a bit magical. Code comments to explain why those numbers were chosen or descriptive variable names would be helpful to make sure someone doesn't screw it up in the future :)

Reply:
Added a variable and a comment to clarify it.